src/libcore/slice.rs

```diff
@@ -2177,6 +2177,16 @@ pub unsafe fn from_raw_parts_mut<'a, T>(p: *mut T, len: usize) -> &'a mut [T] {
 // Comparison traits
 //
 
+// A compile time optimization for faster bytewise slice comparison
+include!(concat!(env!("OUT_DIR"), "/memcmp_optimization.rs"));
+
+// The compare macro for the bytewise slice comparison
+macro_rules! cmp (
+    ($left:expr, $right: expr, $var:ident, $offset:expr) => {{
+        unsafe {*($left.offset($offset) as *const $var) == *($right.offset($offset) as *const $var)}
```

Comment:
Fixing this issue will likely have an impact on the performance. In order to get the best possible performance on platforms which allow unaligned memory access we might need/want to perform this optimisation at a lower level (i.e. replace the implementation of `memcmp` and/or the strategy used in LLVM to replace `memcmp` with inline comparisons).